source/jquery.flot.drawSeries.js

```diff
@@ -533,17 +533,32 @@ This plugin is used by flot for drawing lines, plots, bars or area.
                     barLeft = 0;
                     break;
                 case "right":
-                    barLeft = -series.bars.barWidth;
+                    if (typeof series.bars.barWidth === "number") {
+                        barLeft = -series.bars.barWidth;
+                    } else {
+                        barLeft = -series.bars.barWidth[0];
+                    }
                     break;
                 default:
-                    barLeft = -series.bars.barWidth / 2;
+                    if (typeof series.bars.barWidth === "number") {
+                        barLeft = -series.bars.barWidth / 2;
+                    } else {
+                        barLeft = -series.bars.barWidth[0] / 2;
+                    }
+            }
+
+            var barWidth;
```

Comment:
If you move this above the `switch`, it would greatly simplify all the `barLeft` assignments (i.e. they could just use `barWidth`, instead of repeatedly retrieving it from `series.bars` in each case).